### PR TITLE
Add C implementation for `@stdlib/math/base/special/acosh`

### LINKED ISSUE #641: [RFC]: Add C implementation for `@stdlib/math/base/special/log1p`
### Description

This RFC proposes adding a C implementation for [`@stdlib/math/base/special/log1p`](https://github.com/stdlib-js/stdlib/tree/b43a75923e47775446d3b6d7219ee42e07f7ce63/lib/node_modules/%40stdlib/math/base/special/log1p).

```c
double stdlib_base_log1p( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### LINKED ISSUE #690: [RFC]: Add C implementation for `@stdlib/math/base/special/acosh`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/acosh`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/acosh)
```c
double stdlib_base_acosh( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #690.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a final review.

File: lib/node_modules/@stdlib/math/base/special/acosh/README.md [hunk 1]
```diff
@@ -78,6 +78,93 @@ for ( i = 0; i < x.length; i++ ) {
...
+```
+
+#### stdlib_base_acosh( x )
+
+Computes the [hyperbolic arccosine][hyperbolic-arccosine] of `x`.
+
+```c
+double out = stdlib_base_acosh( 1.0 );
+// returns 0.0
+out = stdlib_base_acosh( 2.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acosh/README.md:114]:
```suggestion
out = stdlib_base_acosh( 2.0 );

```

File: lib/node_modules/@stdlib/math/base/special/acosh/README.md [hunk 2]
```diff
@@ -78,6 +78,93 @@ for ( i = 0; i < x.length; i++ ) {
...
+#include "stdlib/math/base/special/acosh.h"
+```
+
+#### stdlib_base_acosh( x )
+
+Computes the [hyperbolic arccosine][hyperbolic-arccosine] of `x`.
+
+```c
+double out = stdlib_base_acosh( 1.0 );
+// returns 0.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acosh/README.md:113]:
```suggestion
// returns 0.0

```

File: lib/node_modules/@stdlib/math/base/special/acosh/README.md [hunk 3]
```diff
@@ -78,6 +78,93 @@ for ( i = 0; i < x.length; i++ ) {
...
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/acosh.h"
+#include <stdlib.h>
+#include <stdio.h>
+int main() {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acosh/README.md:148]:
```suggestion

int main() {
```

File: lib/node_modules/@stdlib/math/base/special/acosh/README.md [hunk 4]
```diff
@@ -78,6 +78,93 @@ for ( i = 0; i < x.length; i++ ) {
...
+    double x[] = { 1, 1.45, 1.89, 2.33, 2.78, 3.22, 3.66, 4.11, 4.55, 5 };
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acosh/README.md:149]:
```suggestion
    double x[] = { 1.0, 1.45, 1.89, 2.33, 2.78, 3.22, 3.66, 4.11, 4.55, 5.0 };
```

File: lib/node_modules/@stdlib/math/base/special/acosh/benchmark/c/native/benchmark.c [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+int main( void ) {
+	double elapsed;
+	int i;
+
+	// Use the current time to seed the random number generator:
+	srand( time( NULL ) );
+
+	print_version();
+	for ( i = 0; i < REPEATS; i++ ) {
+		printf( "# c::%s\n", NAME );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acosh/benchmark/c/native/benchmark.c:130]:
```suggestion
		printf( "# c::native::%s\n", NAME );
```

File: lib/node_modules/@stdlib/math/base/special/acosh/examples/c/example.c [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/acosh.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x[] = { 1, 1.45, 1.89, 2.33, 2.78, 3.22, 3.66, 4.11, 4.55, 5 };
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acosh/examples/c/example.c:24]:
```suggestion
    double x[] = { 1.0, 1.45, 1.89, 2.33, 2.78, 3.22, 3.66, 4.11, 4.55, 5.0 };
```

**Review by kgryte — APPROVED** [review]
As the changes are minor, I'll merge and touch up on my end. Thanks, @Pranavchiku!

**Pull request merged**